src/platforms/windows/daemon/windowsfirewall.cpp

```diff
@@ -992,3 +993,82 @@ void WindowsFirewall::importAddress(const QHostAddress& addr,
   auto v6bytes = addr.toIPv6Address();
   RtlCopyMemory(&v6bytes, value.byteArray16, IPV6_ADDRESS_SIZE);
 }
+
+
+bool WindowsFirewall::blockTrafficOnPort(uint port, uint8_t weight){
+  logger.debug() << "Requesting to block all traffic on port "<< port;
+
+  auto result = FwpmTransactionBegin(m_sessionHandle, NULL);
+  auto cleanup = qScopeGuard([&] {
+    if (result != ERROR_SUCCESS) {
+      FwpmTransactionAbort0(m_sessionHandle);
+    }
+  });
+  if (result != ERROR_SUCCESS) {
+    logger.error() << "FwpmTransactionBegin0 failed. Return value:.\n"
+                   << result;
+    return false;
+  }
+  // Allow Traffic to IP with PORT using any protocol
+  FWPM_FILTER_CONDITION0 conds[3];
+  conds[0].fieldKey = FWPM_CONDITION_IP_PROTOCOL;
+  conds[0].matchType = FWP_MATCH_EQUAL;
+  conds[0].conditionValue.type = FWP_UINT8;
+  conds[0].conditionValue.uint8 = (IPPROTO_UDP);
+
+  conds[1].fieldKey = FWPM_CONDITION_IP_PROTOCOL;
+  conds[1].matchType = FWP_MATCH_EQUAL;
+  conds[1].conditionValue.type = FWP_UINT8;
+  conds[1].conditionValue.uint16 = (IPPROTO_TCP);
```

Comment:
If the type is `FWP_UINT8` then this should be `conds[1].conditionValue.uint8 = (IPPROTO_TCP)`